source/base/DateTime.ooc

```diff
@@ -14,22 +14,266 @@
 * You should have received a copy of the GNU Lesser General Public License
 * along with this software. If not, see <http://www.gnu.org/licenses/>.
 */
-TimeSpan: cover {
-	_ticks: Int64
-	ticks ::= this _ticks
-	init: func@ (=_ticks)
-	kean_base_timeSpan_getTicks: unmangled func -> Int64 { this _ticks }
-	new: unmangled(kean_base_timeSpan_new) static func ~API (ticks: UInt64) -> This { This new(ticks) }
+
+use ooc-base
+import os/Time
+
+DateTimeData: cover {
+	year: Int { get set }
+	month: Int { get set }
+	day: Int { get set }
+	hour: Int { get set }
+	minute: Int { get set }
+	second: Int { get set }
+	millisecond: Int { get set }
+  init: func@ ()
+	init: func@ ~fromDateTime (=year, =month, =day, =hour, =minute, =second, =millisecond)
+}
+
+extend Time {
+	currentDateTime: static func -> DateTime {
+		result := DateTime new(0)
+		version(windows) {
+				st: SystemTime
+				GetLocalTime(st&)
+				result = DateTime new(st wYear, st wMonth, st wDay, st wHour, st wMinute, st wSecond, st wMilliseconds)
+		}
+		version(!windows) {
+				tt := time(null)
+				tv : TimeVal
+				val := localtime(tt&)
+				gettimeofday(tv&, null)
```

Comment:
It seems `gettimeofday` is marked obsolete anyway. Maybe we can use `clock_gettime` instead?
http://stackoverflow.com/questions/3756323/getting-the-current-time-in-milliseconds